CdnEngine_S3.php

```diff
@@ -384,13 +384,20 @@ public function test( &$error ) {
 			throw new \Exception( 'Bucket doesn\'t exist: %s.', $this->_config['bucket'] );
 		}
 
-		$result = $this->api->putObject( array(
-				'ACL' => 'public-read',
+		if ( ! empty($this->_config['s3acl'])) {
+			$result = $this->api->putObject( array(
+				'ACL' => $this->_config['s3acl'],
```

Comment:
Instead of duplicating putObject here, would passing and empty `$this->_config['s3acl']` work (I did not test this myself yet)?

Reply:
That would mean you'd have to check for an empty value of `ACL` instead of the existence of the key. IMHO I think that's less clean, but I'm not very familiar with PHP standards to happy to make this twek if its more consistent with use of the language.